src/test/kotlin/org/taktik/icure/asynclogic/codelogic/CodeLogicListCodeIdsByTypeCodeVersionIntervalTest.kt

```diff
@@ -0,0 +1,122 @@
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.taktik.icure.asynclogic.CodeLogic
+import org.taktik.icure.services.external.rest.v1.mapper.base.CodeMapper
+import org.taktik.icure.test.CodeBatchGenerator
+import org.taktik.icure.test.ICureTestApplication
+import org.taktik.icure.test.removeEntities
+
+@SpringBootTest(
+	classes = [ICureTestApplication::class],
+	properties = ["spring.main.allow-bean-definition-overriding=true"],
+	webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+@ActiveProfiles("app")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CodeLogicListCodeIdsByTypeCodeVersionIntervalTest @Autowired constructor(
+	private val codeLogic: CodeLogic,
+	private val codeMapper: CodeMapper
+) {
+
+	private val testBatchSize = 100
+	private val codeGenerator = CodeBatchGenerator()
+	private val testBatch = codeGenerator.createBatchOfUniqueCodes(testBatchSize).associateBy { it.id }
+	private val testBatchIds = testBatch.keys.toSortedSet().toList()
+
+	@BeforeAll
+	fun importTestCodeBatch() {
+		runBlocking {
+			testBatch.values.forEach {
+				codeLogic.create(codeMapper.map(it))
+			}
+		}
+	}
+
+	@Test
+	fun allTheIdsAreReturnedIfBothKeysAreNull() {
+		runBlocking {
+			val idsCount = codeLogic.listCodeIdsByTypeCodeVersionInterval(null, null, null, null, null, null)
+				.fold(0) { acc, it ->
+					assert(testBatchIds.contains(it))
+					acc + 1
+				}
+			assertEquals(testBatchIds.size, idsCount)
+		}
+	}
+
+	@Test
+	fun ifStartKeyIsSpecifiedOnlyResultsThatComeAfterAreReturned() {
+		runBlocking {
+			val startIndex = nextInt(0, testBatchIds.size)
+			val startCode = testBatch[testBatchIds[startIndex]]!!
+			val idsCount = codeLogic.listCodeIdsByTypeCodeVersionInterval(startCode.type, startCode.code, startCode.version, null, null, null)
+				.fold(0) { acc, it ->
+					assert(testBatchIds.contains(it))
+					acc + 1
+				}
+			assertEquals(testBatchIds.size - startIndex, idsCount)
+		}
+	}
+
+	@Test
+	fun ifEndKeyIsSpecifiedOnlyResultsThatComeAfterAreReturned() {
+		runBlocking {
+			val endIndex = nextInt(0, testBatchIds.size)
+			val endCode = testBatch[testBatchIds[endIndex]]!!
+			val idsCount = codeLogic.listCodeIdsByTypeCodeVersionInterval(null, null, null, endCode.type, endCode.code, endCode.version)
+				.fold(0) { acc, it ->
+					assert(testBatchIds.contains(it))
+					acc + 1
+				}
+			assertEquals(endIndex + 1, idsCount)
+		}
+	}
+
+	@Test
+	fun ifStartKeyAndEndKeyAreSpecifiedAllTheInBetweenCodesAreReturned() {
+		runBlocking {
+			val startIndex = nextInt(0, testBatchIds.size / 2)
+			val endIndex = nextInt(testBatchIds.size / 2, testBatchIds.size)
+			val startCode = testBatch[testBatchIds[startIndex]]!!
+			val endCode = testBatch[testBatchIds[endIndex]]!!
+			val idsCount = codeLogic.listCodeIdsByTypeCodeVersionInterval(startCode.type, startCode.code, startCode.version, endCode.type, endCode.code, endCode.version)
```

Comment:
I'm wondering what is returned in that one. With an example, if we have the following codes in database : 
- BE-ALLERGEN|226723006|1
- BE-ALLERGEN|226723006|2
- BE-ALLERGEN|412068007|1
- BE-ALLERGEN|412068007|2
- BE-ALLERGEN|412068007|3
- CD-ITEM|allergy|1
- CD-ITEM|allergy|2
- CD-ITEM|allergy|3
- CD-ITEM|evolution|1
- CD-ITEM-EBIRTH|HIVdiagnose|1

If I call `.listCodeIdsByTypeCodeVersionInterval('BE-ALLERGEN', '000000', '1', 'CD-ITEM', 'allergy', '2')`, the service will return : 
- BE-ALLERGEN|226723006|1
- BE-ALLERGEN|226723006|2
- BE-ALLERGEN|412068007|1
- BE-ALLERGEN|412068007|2
- CD-ITEM|allergy|1
- CD-ITEM|allergy|2

correct ?